File: P1 2019.1/Questao(1)/java/Q1_2019_1.java
import java.util.List;
import java.util.Scanner;

import entities.Produto;
import services.Utils;
import services.Utils2;

public class Q1_2019_1 {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        List<Object> listaObjetos = Utils2.getLista();
        String id = sc.nextLine();
        Produto produtoProcurado = new Produto(id);

        if (Utils.existe(listaObjetos, produtoProcurado)) {
            System.out.println(produtoProcurado);
        } else {
            System.out.println("Produto não existe no sistema");
        }

        Produto[] arrayProdutos = transformaListEmArray(listaObjetos);

        double somaProdutos = somaProdutos(arrayProdutos);

        System.out.println(somaProdutos);

        sc.close();
    }

    private static double somaProdutos(Produto[] arrayProdutos) {
        double soma = 0.0;
        for (Produto produto : arrayProdutos) {
            soma += produto.getPreco();
        }
        return soma;
    }

    private static Produto[] transformaListEmArray(List<Object> k) {
        Produto[] produtos;
        int j = 0;
        for (int i = 0; i < k.size(); i++) {
            if (k.get(i) instanceof Produto) {
                j++;
            }
        }
        produtos = new Produto[j];
        j = 0;
        for (int i = 0; i < k.size(); i++) {
            if (k.get(i) instanceof Produto) {
                produtos[j++] = (Produto) k.get(i);
            }
        }
        return produtos;
    }
}

File: P1 2019.1/Questao(1)/python/Q1_2019_1.py
import numpy

from entities.Utils import Utils, Utils2
from entities.Produto import Produto


def main():
    lista_itens = Utils2.get_lista()

    id_produto = input("Id produto para buscar: ")
    produto_buscado = Produto(id_produto)

    if Utils.existe(lista_itens, produto_buscado):
        indice_produto = lista_itens.index(produto_buscado)

        print(lista_itens[indice_produto])

    array_produtos = transforma_list_em_array(lista_itens)

    total_produtos_preco = soma_valor_produtos(array_produtos)

    print(total_produtos_preco)


def transforma_list_em_array(k: list) -> numpy.array(Produto):
    qt = int()
    for o in k:
        qt += (1 if isinstance(o, Produto) else 0)

    array_produtos = numpy.empty(qt, dtype=Produto)
    qt = 0
    for o in k:
        if isinstance(o, Produto):
            array_produtos[qt] = o
            qt += 1

    return array_produtos


def soma_valor_produtos(array_produtos: numpy.array(Produto)) -> float:
    total = float()
    for p in array_produtos:
        total += p.preco

    return total


if __name__ == "__main__":
    main()
